### PR TITLE
feat: add RFC for Ratings component

### PR DESCRIPTION
**Pull request by austin94**
#### Documentation PR Link

This PR introduces the RFC for the `Ratings` component. It is currently a work-in-progress 🚧 

Basic API...

```jsx
import {Rating, KIND} from 'baseui/rating';

<Rating
        kind={KIND.EMOTICON}
        value={this.state.rating}
        onChange={value => {
          this.setState({ rating: value });
        }} 
/>
```

#### License

MIT

### CODE REVIEW

**Comment by austin94** [comment]
@nadiia is there a specific example of documenting the `StatefulContainer` model in an RFC that I should base this off of?

**Comment by nadiia** [comment]
> @nadiia is there a specific example of documenting the `StatefulContainer` model in an RFC that I should base this off of?

[This RFC](https://github.com/uber-web/baseui/blob/master/rfcs/input-component.md) has stateful container API documented

File: rfcs/rating-component.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  state = {
+    rating: 1
+  };
+
+  render() {
+    return (
+      <Rating
+        kind={KIND.EMOTICON}
+        value={this.state.rating}
+        onChange={value => {
```
Comment (nadiia) [inline rfcs/rating-component.md]: should be `onChange={({value}) => {}}`

File: rfcs/rating-component.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+* `STATE_CHANGE_TYPE`
+* `KIND`
+
+## `Rating` API
+
+* `kind: KIND` - Optional
+  The type of `Rating` to display
+* `value: number` - Optional
+  The current rating value
+* `onChange: () => value` - Optional
```
Comment (nadiia) [inline rfcs/rating-component.md]: `onChange: ({value}) => void`

File: rfcs/rating-component.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* `kind: KIND` - Optional
+  The type of `Rating` to display
+* `value: number` - Optional
+  The current rating value
+* `onChange: ({value}) => void` - Optional
+  Callback that returns a newly selected value
+* `overrides: {}` - Optional
+  * `Root: ?React.ComponentType` component to use for root Rating styling
+  * `Star: ?React.ComponentType` component to use for star ratings
+  * `Emoticon: ?React.ComponentType` component to use for emoticon ratings
```
Comment (chasestarr) [inline rfcs/rating-component.md]: if emoticons are override-able, does the index (or explicit enum) need to be provided?

Comment (austin94) [reply rfcs/rating-component.md]: I've added 2 new props to the styled-components to address this. `$isSelected` and `$index`

**Pull request merged**